storage/innobase/handler/ha_innodb.cc

```diff
@@ -1776,30 +1776,29 @@ static void innodb_space_shutdown() {
 static int innodb_shutdown(handlerton *, ha_panic_function) {
   DBUG_TRACE;
 
-  if (innodb_inited) {
-    innodb_inited = false;
-    hash_table_free(innobase_open_tables);
-    innobase_open_tables = nullptr;
+  hash_table_free(innobase_open_tables);
+  innobase_open_tables = nullptr;
 
-    for (auto file : innobase_sys_files) {
-      ut::delete_(file);
-    }
-    innobase_sys_files.clear();
-    innobase_sys_files.shrink_to_fit();
+  for (auto file : innobase_sys_files) {
+    ut::delete_(file);
+  }
+  innobase_sys_files.clear();
+  innobase_sys_files.shrink_to_fit();
 
-    mutex_free(&master_key_id_mutex);
-    srv_shutdown();
-    innodb_space_shutdown();
+  if (mutex_monitor) mutex_free(&master_key_id_mutex);
```

Comment:
This is create at the very start. IMHO, we shouldn't need the mutex_monitor check

Reply:
Returns from `srv_start()` exist above `master_key_id_mutex` creating. `if` need

Reply:
You mean, we can reach here without srv_start()?

Reply:
See https://github.com/tempesta-tech/percona-server/blob/avb-log_file_name/storage/innobase/srv/srv0start.cc#L2192
This return before `master_key_id_mutex` creating